dds/DCPS/ValueCommon.h

```diff
@@ -0,0 +1,173 @@
+  {
+    for (const Pair* ptr = pairs; ptr->name; ++ptr) {
+      pairs_.push_back(*ptr);
+    }
+  }
+
+  void pairs(const OPENDDS_VECTOR(Pair)& pairs)
+  {
+    pairs_ = pairs;
+  }
+
+  bool get_value(ACE_CDR::Long& value, const char* name) const
+  {
+    for (OPENDDS_VECTOR(Pair)::const_iterator it = pairs_.begin(); it != pairs_.end(); ++it) {
+      if (std::strcmp(it->name, name) == 0) {
+        value = it->value;
+        return true;
+      }
+    }
+    return false;
+  }
+
+  bool get_name(const char*& name, ACE_CDR::Long value) const
+  {
+    for (OPENDDS_VECTOR(Pair)::const_iterator it = pairs_.begin(); it != pairs_.end(); ++it) {
+      if (it->value == value) {
+        name = it->name;
+        return true;
+      }
+    }
+    return false;
+  }
+
+  XTypes::TypeKind get_equivalent_int() const
+  {
+    return as_int_;
+  }
+
+private:
+  OPENDDS_VECTOR(Pair) pairs_;
+  XTypes::TypeKind as_int_;
+};
+
+class BitmaskHelper {
+public:
+  virtual ~BitmaskHelper() {}
+  virtual bool get_value(ACE_CDR::ULongLong& value, const OPENDDS_VECTOR(const char*)& names) const = 0;
+  virtual size_t get_names(OPENDDS_VECTOR(const char*)& names, ACE_CDR::ULongLong value) const = 0;
+  virtual XTypes::TypeKind get_equivalent_uint() const = 0;
+};
+
+class MapBitmaskHelper : public BitmaskHelper {
+public:
+  struct Pair {
+    const char* name;
+    ACE_CDR::UShort position;
+  };
+
+  typedef OPENDDS_MAP(ACE_CDR::UShort, const char*) PosToNameMap;
+  typedef OPENDDS_MAP(const char*, ACE_CDR::UShort) NameToPosMap;
+  typedef PosToNameMap::const_iterator ptn_iterator;
+  typedef NameToPosMap::const_iterator ntp_iterator;
+
+  MapBitmaskHelper(ACE_CDR::UShort bit_bound, XTypes::TypeKind as_uint)
+    : bit_bound_(bit_bound)
+    , as_uint_(as_uint)
+  {}
+
+  MapBitmaskHelper(const Pair* pairs, ACE_CDR::UShort bit_bound, XTypes::TypeKind as_uint)
+    : bit_bound_(bit_bound)
+    , as_uint_(as_uint)
+  {
+    for (const Pair* ptr = pairs; ptr->name; ++ptr) {
+      pos_to_name_.insert(std::make_pair(ptr->position, ptr->name));
+      name_to_pos_.insert(std::make_pair(ptr->name, ptr->position));
+    }
+  }
+
+  void pairs(const OPENDDS_VECTOR(Pair)& pairs)
+  {
+    for (OPENDDS_VECTOR(Pair)::const_iterator it = pairs.begin(); it != pairs.end(); ++it) {
+      pos_to_name_.insert(std::make_pair(it->position, it->name));
+      name_to_pos_.insert(std::make_pair(it->name, it->position));
+    }
+  }
+
+  bool get_value(ACE_CDR::ULongLong& value, const OPENDDS_VECTOR(const char*)& names) const
+  {
+    ACE_CDR::ULongLong rtn = 0;
+    for (size_t i = 0; i < names.size(); ++i) {
+      const ntp_iterator it = name_to_pos_.find(names.at(i));
+      if (it == name_to_pos_.end()) {
+        return false;
+      }
+      rtn |= 1 << it->second;
+    }
+    value = rtn;
+    return true;
+  }
+
+  // Return an estimated length of a string constructed by the returned flag names
+  // with a delimiter character between two consecutive flags.
```

Comment:
Move this comment to the base class definition (assuming it applies to the interface and not the implementation)